src/test/ui/rfc-1260/reexport_from_extern_crate.rs

```diff
@@ -0,0 +1,7 @@
+// run-pass
+// aux-build:main_functions.rs
+
+#![feature(main_by_path)]
+
+extern crate main_functions;
+pub use main_functions::boilerplate as main;
```

Comment:
In this case, does `boilerplate` have a corresponding `LocalDefId`? Or do i need to refactor existing code to use `DefId`? @petrochenkov

Reply:
`DefId` needs to be used.

Reply:
Then i need to tweak a few bits of typeck... The current implementation retrieves hir from `localdefid` and check its type. Is there anything equivalent for `defid` ?